src/mlpack/methods/ann/layer/separable_convolution.hpp

```diff
@@ -0,0 +1,430 @@
+/**
+ * @file methods/ann/layer/separable_convolution.hpp
+ * @author Kartik Dutt
+ * @author Aakash Kaushik
+ * @author Sidharth
+ * 
+ * For more information, kindly refer to the following paper.
+ *
+ * @code
+ * @article{
+ *  author = {Laurent SIfre, Stéphane Mallat},
+ *  title = {Rigid-Motion Scattering for Texture Classification},
+ *  year = {2014},
+ *  url = {https://arxiv.org/pdf/1403.1687}
+ * }
+ * @endcode
+ * 
+ * Definition of the Separable Convolution module class.
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SEPARABLE_CONVOLUTION_HPP
+#define MLPACK_METHODS_ANN_LAYER_SEPARABLE_CONVOLUTION_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/ann/convolution_rules/naive_convolution.hpp>
+#include <mlpack/methods/ann/layer/padding.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+
+namespace mlpack {
+
+/**
+ * Implementation of the Separable Convolution class.The Separable Convolution class 
+ * represents a single layer of a neural network.
```

Comment:
This comment doesn't really explain much; do you think you can clarify it?

Reply:
please check if the new comment is better and should we add something else too

Reply:
Thanks, that is indeed much improved!